fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/group/ChunkReader.java

```diff
@@ -0,0 +1,328 @@
+
+                if (fhirSearchFromDate != null) {
+                    searchCreterial.add("ge" + fhirSearchFromDate);
+                }
+                if (fhirSearchToDate != null) {
+                    searchCreterial.add("lt" + fhirSearchToDate);
+                }
+                if (!searchCreterial.isEmpty()) {
+                    queryParameters.put(Constants.FHIR_SEARCH_LASTUPDATED, searchCreterial);
+                }
+
+                queryParameters.put("_sort", Arrays.asList(new String[] { Constants.FHIR_SEARCH_LASTUPDATED }));
+                searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                        ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, null, true);
+                do {
+                    searchContext.setPageSize(pageSize);
+                    searchContext.setPageNumber(compartmentPageNum);
+                    FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                    txn.begin();
+                    FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                    List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                    txn.commit();
+                    compartmentPageNum++;
+
+                    for (Resource res2 : resources2) {
+                        if (res2 == null) {
+                            continue;
+                        }
+                        try {
+                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
+                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR.getBytes());
+                            resSubTotal++;
+                        } catch (FHIRGeneratorException e) {
+                            logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
+                                        + patientId + "'", e);
+                        } catch (IOException e) {
+                            logger.warning("fillChunkDataBuffer: chunkDataBuffer written error!");
+                            throw e;
+                        }
+                    }
+
+                } while (searchContext.getLastPageNumber() >= compartmentPageNum);
+            }
+            chunkData.setCurrentPartResourceNum(chunkData.getCurrentPartResourceNum() + resSubTotal);
+            logger.fine("fillChunkDataBuffer: Processed resources - " + resSubTotal + "; Bufferred data size - "
+                    + chunkData.getBufferStream().size());
+        } else {
+            logger.warning("fillChunkDataBuffer: chunkData is null, this should never happen!");
+            throw new Exception("fillChunkDataBuffer: chunkData is null, this should never happen!");
```

Comment:
shouldn't this throw a more specific exception?  more easily differentiating in code downstream?

Reply:
not needed, the job always exit when this rare error happens, exception info in log is enough